components/backup-stream/src/observer.rs

```diff
@@ -18,20 +17,6 @@ use crate::{
     utils::SegmentSet,
 };
 
-/// The inflight `StartObserve` message count.
-/// Currently, we handle the `StartObserve` message in the main loop(endpoint
-/// thread), which may take longer time than expected. So when we are starting
-/// to observe many region (e.g. failover), there may be many pending messages,
-/// those messages won't block the advancing of checkpoint ts. So the checkpoint
-/// ts may be too late and losing some data.
-///
-/// This is a temporary solution for this problem: If this greater than (1),
-/// then it implies that there are some inflight wait-for-initialized regions,
-/// we should block the resolved ts from advancing in that condition.
-///
-/// FIXME: Move handler of `ModifyObserve` to another thread, and remove this :(
-pub static IN_FLIGHT_START_OBSERVE_MESSAGE: AtomicUsize = AtomicUsize::new(0);
```

Comment:
For reviewers: this variable isn't read anymore. We have already get rid of it by advancing the resolved TS after syncing the pending messages in the queue. The reason this variable is still here is that I forgot to remove it once fix that problem. 🥲